aws/aws-kinesis-firehose/main.tf

```diff
@@ -0,0 +1,72 @@
+locals {
+  destinations = {
+    "s3" = "extended_s3"
+  }
+
+  default_key_prefix     = "data/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/"
+  partitioned_key_prefix = "data/!{partitionKeyFromQuery:partition_key}/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/"
+
+  key_prefix = var.dynamic_partitioning_enabled ? local.partitioned_key_prefix : local.default_key_prefix
+  name       = "${var.name}-firehose"
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "main" {
+  name        = local.name
+  destination = local.destinations[var.destination]
+
+  dynamic "extended_s3_configuration" {
+    for_each = var.dynamic_partitioning_enabled ? [1] : []
+    content {
+      role_arn            = aws_iam_role.firehose_role.arn
+      bucket_arn          = var.bucket_arn
+      prefix              = local.key_prefix
+      error_output_prefix = "errors/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/!{firehose:error-output-type}/"
+      buffer_size         = var.buffer_size     # In MB
+      buffer_interval     = var.buffer_interval # In seconds
+
+      dynamic_partitioning_configuration {
+        enabled = true
+      }
+
+      processing_configuration {
+        enabled = true
+
+        processors {
+          type = "MetadataExtraction"
+          parameters {
+            parameter_name  = "JsonParsingEngine"
+            parameter_value = "JQ-1.6"
+          }
+          parameters {
+            parameter_name  = "MetadataExtractionQuery"
+            parameter_value = format("{partition_key:%s}", var.query)
+          }
+        }
+      }
+    }
+  }
+}
+
+resource "aws_iam_role" "firehose_role" {
+  name        = "firehose_test_role"
```

Comment:
```suggestion
  name        = "${local.name}-role"
```

Reply:
I’d say leave off “role” you know it’s a role by the ARN and resource types. We would be using a suffix to distinguish between N roles in the same bundle/mod